_docs/technical/protocols/2017-01-13-binary.md

```diff
@@ -1169,6 +1148,118 @@ data ConsensusData (GenesisBlockchain ssc) = GenesisConsensusData
 |  1-10 | EpochIndex      | Index of epoch for which this genesis block is relevant |
 |  1-10 | ChainDifficulty | Difficulty of the chain ending in this genesis block.   |
 
+### BlockHeader
+
+~~~ haskell
+-- | Either header of ordinary main block or genesis block.
+type BlockHeader ssc = Either (GenesisBlockHeader ssc) (MainBlockHeader ssc)
+~~~
+
+| Tag size | Tag Type | Tag Value | Description                  | Field size               |
+|----------+----------+-----------+------------------------------+--------------------------|
+|        1 | Word8    |      0x00 | Tag for `GenesisBlockHeader` |                          |
+|          |          |           |                              | size(GenesisBlockHeader) |
+|          |          |      0x01 | Tag for `MainBlockHeader`    |                          |
+|          |          |           |                              | size(MainBlockHeader)    |
+
+[//]: TODO: Describe Block
+
+## Block Exchange Messages
+
+### HeaderHash
+
+~~~ haskell
+-- | 'Hash' of block header. This should be @Hash (BlockHeader ssc)@
+-- but we don't want to have @ssc@ in 'HeaderHash' type.
```

Comment:
@rockbmb `ssc` parameter was removed here under this issue: 
https://issues.serokell.io/issue/CSL-604
I don't think reader of documentation interested in reasons why we are not using `HeaderHash` there. But I can fix comment to:

```haskell
-- | 'Hash' of block header. This should be @Hash (BlockHeader ssc)@
-- but there is no @ssc@ in HeaderHash type.
```